test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/Streaming/ODataJsonLightStreamWritingTests.cs

```diff
@@ -67,6 +67,79 @@ public void CanWritePrimitivePropertyValue()
             expectedPayload);
         }
 
+        [Fact]
+        public void CanWritePrimitivePropertyInstanceAnnotationWithoutPropertyValue()
+        {
+            string expectedPayload = String.Format(resourcePayload,
+                ",\"age@Custom.StartAnnotation\":123"
+                );
+
+            RunTest((ODataWriter writer) =>
+            {
+                ODataPrimitiveValue primitiveValue = new ODataPrimitiveValue(123);
+                ODataPropertyInfo propertyInfo = new ODataPropertyInfo { Name = "age" };
+                propertyInfo.InstanceAnnotations.Add(new ODataInstanceAnnotation("Custom.StartAnnotation", primitiveValue));
+
+                // write some properties within the resource and others separately
+                writer.WriteStart(resource);
+
+                writer.WriteStart(propertyInfo);
+                writer.WriteEnd(); // property
+
+                writer.WriteEnd(); // resource
+            },
+            expectedPayload);
+        }
+
+        [Fact]
+        public void CanWritePrimitivePropertyInstanceAnnotationWithPropertyValue()
+        {
+            string expectedPayload = String.Format(resourcePayload,
+                ",\"age@Custom.StartAnnotation\":123,\"age\":37"
+                );
+
+            RunTest((ODataWriter writer) =>
+            {
+                ODataPrimitiveValue primitiveValue = new ODataPrimitiveValue(123);
+                ODataPropertyInfo propertyInfo = new ODataPropertyInfo { Name = "age" };
+                propertyInfo.InstanceAnnotations.Add(new ODataInstanceAnnotation("Custom.StartAnnotation", primitiveValue));
+
+                // write some properties within the resource and others separately
+                writer.WriteStart(resource);
+
+                writer.WriteStart(propertyInfo);
+                    writer.WritePrimitive(new ODataPrimitiveValue(37));

```

Comment:
Fix indentation

Reply:
@xuzhg I don't think this approach is sustainable. The indentation will most probably be undone the next time VS auto-formats the contents of this file

Reply:
auto-format automatically runs? We have lots of such intentionally indent in our test codes, so far, I haven't seen any un-done.

Reply:
I meant that the next contributor who "touches" that file and triggers the formatter (which often happens), the indentation will be undone. Though I understand what you're trying to visually communicate by applying the indentation.

Reply:
We should review the chances and let the contributor to revert it.